server/src/test/java/com/google/coffeehouse/servlets/GetProfileServletTest.java

```diff
@@ -0,0 +1,145 @@
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+/**
+ * Unit tests for {@link GetProfileServlet}.
+ */
+public class GetProfileServletTest {
+  private static final String USER_ID = "predetermined-identification-string";
+  private static final String EMAIL = "email@test.com";
+  private static final String NICKNAME = "test";
+  private static final String PRONOUNS = "they";
+  private static final String JSON = String.join("\n",
+      "{",
+      "  \"" + Person.USER_ID_FIELD_NAME + "\" : \"" + USER_ID + "\"",
+      "}");
+  private static final String NO_USER_ID_JSON = "{}";
+  private static final String PROFILE_NOT_FOUND_JSON = String.join("\n",
+      "{",
+      "  \"" + Person.USER_ID_FIELD_NAME + "\" : \"\"",
+      "}");
+  private static final String SYNTACTICALLY_INCORRECT_JSON = "{\"" + Person.USER_ID_FIELD_NAME + "\"";
+
+  private Person testPerson = Person.newBuilder()
+                                    .setEmail(EMAIL)
+                                    .setNickname(NICKNAME)
+                                    .setUserId(USER_ID)
+                                    .setPronouns(PRONOUNS)
+                                    .build();
+  private GetProfileServlet getProfileServlet;
+  private GetProfileServlet failingGetProfileServlet;
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper();
+  private StringWriter stringWriter = new StringWriter();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  @Mock private StorageHandlerApi successfulHandler;
+  @Mock private StorageHandlerApi failingHandler;
+  
+  @Before
+  public void setUp() throws IOException {
+    helper.setUp();
+
+    successfulHandler = mock(StorageHandlerApi.class);
+    when(successfulHandler.fetchPersonFromId(anyString())).thenReturn(testPerson);
+    getProfileServlet = new GetProfileServlet(successfulHandler);
+
+    failingHandler = mock(StorageHandlerApi.class);
+    when(failingHandler.fetchPersonFromId(anyString()))
+                       .thenThrow(new IllegalArgumentException(
+                                      StorageHandler.PERSON_DOES_NOT_EXIST));
+    failingGetProfileServlet = new GetProfileServlet(failingHandler);
+
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+    when(response.getWriter()).thenReturn(new PrintWriter(stringWriter));
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGet_validInput() throws IOException {
+    when(request.getReader()).thenReturn(
+          new BufferedReader(new StringReader(JSON)));
+
+    getProfileServlet.doGet(request, response);
+    String result = stringWriter.toString();
+
+    Gson gson = new Gson();
+    Person p = gson.fromJson(result, Person.class);
+
+    assertEquals(USER_ID, p.getUserId());
```

Comment:
nit: Can you check other fields too, just to make sure everything is coming in from the DB correctly